src/element/textWysiwyg.tsx

```diff
@@ -173,29 +171,12 @@ export const textWysiwyg = ({
         ? approxLineHeight
         : updatedElement.height / lines.length;
       if (!container) {
-        maxWidth =
-          (appState.offsetLeft + appState.width - viewportX - 8) /
-            appState.zoom.value -
-          // margin-right of parent if any
-          Number(
-            getComputedStyle(
-              excalidrawContainer?.parentNode as Element,
-            ).marginRight.slice(0, -2),
-          );
```

Comment:
note sure about what this solved before, but shouldn't w keep this?

Reply:
I removed this deliberately and after extensive testing. I think it is based on the misunderstanding that appState.width is the entire screen width, while in reality, it is the canvasDOMWidth. Subtracting the right-hand side margin is unnecessary. 

https://github.com/excalidraw/excalidraw/blob/c009e03c8e654f89739f840c5b374bbb9d1d2982/src/components/App.tsx#L380-L395